tuist/tuist
# Add support for dependencies on app targets from different projects

In certain cases, such as when the test host app resides in a different project, it's necessary to depend on an .app target from another project. To enable this, a file reference to the app is required, followed by setting up a dependency on it. This commit introduces file references for all .project executable targets, and adds a build phase to copy the .app file, making it available during the build process.

### Short description 📝

Allow dependency of `.app` targets from another project.

### How to test the changes locally 🧐

Create a project with an app target, and depend on this target in another project. Before the change, there should be no affect, and after the change, there should be a dependency.

For a more realistic flow, create a test host app, and use it in a project with on app targets (fixtures that demonstrate this - TBD)

### Contributor checklist ✅


More tasks:

- [x] Check if this solves this issue: https://github.com/tuist/tuist/issues/6130
- [x] Add fixture with such example

### Reviewer checklist ✅
